client/src/pages/contentPages/Groups.js

```diff
@@ -1,23 +1,275 @@
+      padding: theme.spacing(2, 4, 3),
+      borderRadius: 4,
+      outline: 'none'
+
+    },
+
+    groupImageContainer: {
+      flex: 1,
+      display: 'flex',
+      flexDirection: 'column',
+      justifyContent: 'center',
+      alignItems: 'center',
+      paddingTop: "10px",
+      paddingBottom: "10px"
+    },
+    large: {
+      height: "300px",
+      width: "400px",
+    },
+    uploading: {
+      opacity: 0.5,
+      height: "300px",
+      width: "400px"
+    },
+    
   }));
 
+  function getModalStyle() {
+    const top = 50;
+    const left = 50;
+  
+    return {
+      top: `${top}%`,
+      left: `${left}%`,
+      transform: `translate(-${top}%, -${left}%)`,
+    };
+  }
+
+
+
 const Groups = () => {
     const classes = useStyles();
-    const {userGroups, dispatch} = useGlobalContext();
+    const {userGroups, userCourse, dispatch} = useGlobalContext();
+
     const {groups, courseGroups} = userGroups;
     const [suggestedGroups, setSuggestedGroups] = React.useState([]);
 
+    const { enqueueSnackbar } = useSnackbar();
+    const [modalStyle] = React.useState(getModalStyle);
+    const [openModal, setOpen] = useState(false);
+    const [groupError, setGroupErrors] = useState('');
+    const [uploading, setUploading] = useState(false);
+    const [groupPicture, setGroupPicture] = useState('');
+    const [formValid, setFormValid] = useState(false);
+    const [groupName, setGroupName] = useState('');
+    const [courseId, setCourseId] = useState('');
+
+
+    const onDrop = useCallback(async (droppedFiles) => {
+      if (droppedFiles.length) {
+          setUploading(true);
+  
+          const form = new FormData();
+          form.append('image', droppedFiles[0]);
+          const res = await axios.post('/upload/single', form)
+            .catch((err) => console.log(err));
+          if (res && res.data) {
+            setGroupPicture(res.data.imageUrl)
+            setUploading(false)
+          }
+      }
+    }, []);
+  
+    const {
+      getRootProps,
+      isDragActive,
+      isDragAccept,
+      isDragReject,
+    } = useDropzone({onDrop, maxFiles: 1, accept: 'image/*'});
+  
+    const style = useMemo(() => ({
+      ...baseStyle,
+      ...(isDragActive ? activeStyle : {}),
+      ...(isDragAccept ? acceptStyle : {}),
+      ...(isDragReject ? rejectStyle : {})
+    }), [
+      isDragActive,
+      isDragReject,
+      isDragAccept
+    ]);
+
+    const handleOpen = () => {
+      setOpen(true);
+      setGroupPicture('');
+    };
+  
+    const handleClose = () => {
+      setOpen(false);
+      setGroupPicture('');
+      setGroupName('');
+      setGroupErrors('');
+    };
+    
+    const handleSubmit = async (e) => {
+      e.preventDefault();
+      
+      if(formValid) {
+  
+        const data = { groupName: groupName, imageUrl: groupPicture , courseId: courseId};
+        
+        const response = await axios.post('/user/groups', data);
+
+        setGroupName('');
+  
```

Comment:
You want to do something similar for courseId and groupPicture as well. I mean clear them after creating a group as you did with the group name.